store_with_fallbacks.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials
+
+import (
+	"context"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// StoreWithFallbacks is a store that has multiple fallback stores.
+type StoreWithFallbacks struct {
+	stores []Store
+}
+
+// NewStoreWithFallbacks returns a new store based on the given stores.
+// The second and the subsequent stores will be used as fallbacks for the first store.
+func NewStoreWithFallbacks(store Store, fallbacks ...Store) Store {
```

Comment:
I'm still struggling between

```go
NewStoreWithFallbacks(store Store, fallbacks ...Store) Store
```
and 
```go
NewStoreWithFallbacks(stores ...Store) Store
```


Later we will have a docker store which is based on the docker configuration file at the default location. I'm wondering how should we combine it with `StoreWithFallbacks`.

If we go for option 1, the experience would be:
```go
ns := NewNativeStore()
store := NewStoreWithFallbacks(ns, NewStoreFromDocker())
```
or
```go
dockerStore := NewStoreFromDocker()
store := NewStoreWithFallbacks(dockerStore)
```

If we go for option 2, we should have the docker store to be the default store when no store is passed.
So 
```go
dockerStore := NewStoreFromDocker()
store := NewStoreWithFallbacks(dockerStore)
```
is equivalent to

```go
store := NewStoreWithFallbacks()
```

Reply:
I think I still prefer option 1.